.github/scripts/create-vm.py

```diff
@@ -0,0 +1,465 @@
+import os
+import grpc
+import json
+import requests
+import math
+import logging
+import argparse
+import yaml
+import random
+import string
+from github import Github, Auth as GithubAuth
+from yandexcloud import SDK, RetryInterceptor, backoff_linear_with_jitter
+from yandex.cloud.compute.v1.instance_service_pb2_grpc import InstanceServiceStub
+from yandex.cloud.compute.v1.instance_pb2 import Instance
+from yandex.cloud.compute.v1.instance_service_pb2 import (
+    CreateInstanceRequest,
+    ResourcesSpec,
+    AttachedDiskSpec,
+    NetworkInterfaceSpec,
+    PrimaryAddressSpec,
+    OneToOneNatSpec,
+    DeleteInstanceRequest,
+    CreateInstanceMetadata,
+    DeleteInstanceMetadata,
+)
+from yandex.cloud.compute.v1.instance_pb2 import IpVersion
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s: %(levelname)s: %(message)s"
+)
+
+
+def github_output(key, value, secret=False):
+    GITHUB_OUTPUT = os.environ.get("GITHUB_OUTPUT")
+
+    if GITHUB_OUTPUT:
+        with open(GITHUB_OUTPUT, "a") as fp:
+            fp.write(f"{key}={value}\n")
+
+    logging.info(f"echo \"{key}={'******' if secret else value}\" >> $GITHUB_OUTPUT")
+
+
+def generate_github_label():
+    generated_string = "".join(
+        random.choices(string.ascii_lowercase + string.digits, k=8)
+    )
+    logging.info(f"Generated label: {generated_string}")
+    return generated_string
+
+
+class KeyValueAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):  # noqa: U100
+        kv_dict = {}
+        for item in values.split(","):
+            key, value = item.split("=")
+            kv_dict[key] = value
+        setattr(namespace, self.dest, kv_dict)
+
+
+def fetch_github_team_public_keys(gh, github_org, team_slug):
+    org = gh.get_organization(github_org)
+    team = org.get_team_by_slug(team_slug)
+    members = [member for member in team.get_members()]
+
+    ssh_keys = []
+    logging.info(
+        f"Fetching SSH keys for members: {[member.login for member in members]}"
+    )
+    member_keys_count = 0
+    for member in members:
+
+        for key in member.get_keys():
+            member_keys_count += 1
+            ssh_keys.append(key.key)
+
+        logging.debug(f"Fetched {member_keys_count} SSH keys for {member.login}")
+
+    logging.debug(f"Fetched SSH keys: {ssh_keys}")
+    return ssh_keys
+
+
+def generate_cloud_init_script(user, ssh_keys, owner, repo, token, version, label):
```

Comment:
Please, add a typehint

Reply:
I understand that it is good practice, but I don't see why it would be necessary here. It is not really big script and for now it is not intended to use it outside of our workflows